tests/web/test_json.py

```diff
@@ -233,6 +233,50 @@ def testGuardianResponse(client, version, buildTarget, channel, code, response):
         assert "Rule-Data-Version" in ret.headers
 
 
+@pytest.mark.usefixtures("appconfig", "guardian_db", "disable_errorhandler", "mock_autograph")
+@pytest.mark.parametrize(
+    "version,buildTarget,channel,code,response",
+    [
+        (
+            "0.4.0.0",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/0.5.0.0.msi", "version": "0.5.0.0", "hashFunction": "sha512", "hashValue": "abcdef"},
+        ),
+        (
+            "0.6.0.0",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/1.0.0.0.msi", "version": "1.0.0.0", "hashFunction": "sha512", "hashValue": "mnopqr"},
+        ),
+        (
+            "0.99.99.99",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/1.0.0.0.msi", "version": "1.0.0.0", "hashFunction": "sha512", "hashValue": "mnopqr"},
+        ),
+        ("1.0.0.0", "WINNT_x86_64", "release", 404, {}),
+        ("0.6.0.0", "Linux_x86_64", "release", 404, {}),
+        ("0.6.0.0", "WINNT_x86_64", "beta", 404, {}),
+        # This shouldn't match because the rule on the alpha channel contains fields not used by this type of update query.
+        ("0.6.0.0", "WINNT_x86_64", "alpha", 404, {}),
+        ("0.6.0.0", "Darwin_x86_64", "evilrelease", 200, {}),
+    ],
+)
+def testGuardianResponseWithCanonicalJson(client, version, buildTarget, channel, code, response):
+    ret1 = client.get(f"/json/1/Guardian/{version}/{buildTarget}/{channel}/update.json")
+    ret2 = client.get(f"/json/1/Guardian/{version}/{buildTarget}/{channel}/update.json")
```

Comment:
I'm not entirely certain what you're testing here. You seem to make making the same request twice, and asserting that the results are the same (which I would assume they would be...).

I think what you _should_ be doing is comparing a string version of the JSON with a string that is hardcoded in the expected, canonical order.

Reply:
I have made the requested change.